src/Geometry.cpp

```diff
@@ -16,510 +14,744 @@
-  const Json::Value& full_hexagons_area_module = hexagons["FH_area_module"]; 
-  const Json::Value& full_hexagons_count = hexagons["FH_count"]; 
-  // FIXME: should add more checks
-  if(debug) {
-    std::cout << "Full hexagon area " << full_hexagons_area.asDouble() << std::endl;
-    std::cout << "Full hexagon area module " << full_hexagons_area_module.asDouble() << std::endl;
-    std:: cout << "Full hexagons nbr of cells " << full_hexagons_count.asUInt() << std::endl;
-    std::cout << " " << std::endl;
-  }
-
-  // construct full hexagon cells
-  for (unsigned int icell=0; icell<full_hexagons_count.asUInt(); icell++) {
-    std::string cell_name = fixedLength(icell,5,"FH");
-    // FIXME: add more checks
-    const Json::Value& hexagon_attributes = hexagons[cell_name]; 
-    int i_index = hexagon_attributes[0]["mapping_coord"][0].asInt();
-    int j_index = hexagon_attributes[0]["mapping_coord"][1].asInt();
-    TVectorD position(3);
-    position(0) = hexagon_attributes[1]["center_coord"][0].asDouble()/10.; // cm
-    position(1) = hexagon_attributes[1]["center_coord"][1].asDouble()/10.; // cm
-    position(2) = getZlayer();
-    double orientation = hexagon_attributes[2]["orientation"].asDouble();
-    std::vector<TVectorD> vertices;
-    for (unsigned i=0; i<hexagon_attributes[3]["vertex_coord_abs"].size(); i++) {
-      vertices.emplace_back(2);
-      vertices.back()(0) = hexagon_attributes[3]["vertex_coord_abs"][i][0].asDouble()/10.; // cm
-      vertices.back()(1) = hexagon_attributes[3]["vertex_coord_abs"][i][1].asDouble()/10.; // cm
+    reverse(result.begin(), result.end());
+    result = type + result;
+    return result;
+}
+
 
+void Geometry::setLayer(int klayer) {
+    if (klayer<0 || unsigned(klayer)>=parameters_.layers_z.size()) {
+        stringstream error;
+        error << "[Geometry] error, invalid klayer " << klayer;
+        throw error.str();
     }
-    cells_.emplace(
-        Cell::id(i_index, j_index),
-        Cell(std::move(position), std::move(vertices), orientation, i_index, j_index)
-        );
+    klayer_ = klayer;
+}
 
-    if(debug) {
-      std::cout << "New cell " << cell_name << " : " << std::endl;
-      std::cout << " mapping coordinates : " << 
-        hexagon_attributes[0]["mapping_coord"][0].asInt() << " " <<
-        hexagon_attributes[0]["mapping_coord"][1].asInt() << std::endl;
-      std::cout << " center coordinates : " << 
-        hexagon_attributes[1]["center_coord"][0].asDouble() << " " <<
-        hexagon_attributes[1]["center_coord"][1].asDouble() << std::endl;
-      std::cout << " orientation : " << 
-        hexagon_attributes[2]["orientation"].asDouble() << std::endl;
-      std::cout << " vertex_coordinates : " << std::endl;
-      for(const auto& vertex : vertices) {
-        std::cout << "  vertex " << 
-          vertex(0) << " " <<
-          vertex(1) << std::endl;
-      }
+
+string Geometry::setHgcalPart(int klayer) {
+    if (klayer < 28)
+        return "EE";
```

Comment:
Better to use brackets around `if` `else`

Reply:
Fixed